### ISSUE TITLE
False positive "unknown-argument" with pydantic and mixin

### ISSUE THREAD

**Issue by a14n** [astral-sh/ty#3861]
### Summary

`ty` doesn't seem to handle fields on mixins with pydantic.

```py
from pydantic import BaseModel, Field


class Mixin:
    mixin_field: bool = Field(default=...)


class MyModel(BaseModel, Mixin):
    model_field: bool = Field(default=...)


if __name__ == "__main__":
    print(MyModel(model_field=True, mixin_field=False).model_dump())
```

Runing this file works but `ty` triggers an error:

```
{'mixin_field': False, 'model_field': True}
```

Here's the `uvx ty check ty.py` (`ty 0.0.54 (bf8a7dd58 2026-06-25)`) output:

```
error[unknown-argument]: Argument `mixin_field` does not match any known parameter
  --> ty.py:13:37
   |
13 |     print(MyModel(model_field=True, mixin_field=False).model_dump())
   |                                     ^^^^^^^^^^^^^^^^^
   |
```

### Version

ty 0.0.54 (bf8a7dd58 2026-06-25)

**Comment by sharkdp** [comment]
Hm, interesting that this is allowed in pydantic even if `Mixin` is not a model itself. We currently don't have special support for pydantic. I'll add this as a child issue of #2403.

In the meantime, you could use the following to make this work with both pydantic (hopefully) and ty. Note the reversed base class order on `MyModel`

```py
from pydantic import BaseModel, Field


class Mixin(BaseModel):
    mixin_field: bool = Field(default=...)


class MyModel(Mixin, BaseModel):
    model_field: bool = Field(default=...)


if __name__ == "__main__":
    print(MyModel(model_field=True, mixin_field=False).model_dump())```
